python/lsst/meas/extensions/gaap/_gaussianizePsf.py

```diff
@@ -0,0 +1,370 @@
+                3) padPsfBy to >= %s
+                """ % (self.kConfig.kernelSize, psfSize,
+                       maxKernelSize, self.kConfig.kernelSize - maxKernelSize)
+            raise ValueError(message)
+
+        dimenS = geom.Extent2I(psfSize, psfSize)
+
+        if (dimenR != dimenS):
+            try:
+                targetPsfModel = targetPsfModel.resized(psfSize, psfSize)
+            except Exception as e:
+                self.log.warn("Zero padding or clipping the target PSF model of type %s and dimensions %s"
+                              " to the science Psf dimensions %s because: %s",
+                              targetPsfModel.__class__.__name__, dimenR, dimenS, e)
+            dimenR = dimenS
+
+        # Make the target kernel image, at location of science subimage.
+        targetMI = self._makePsfMaskedImage(targetPsfModel, center, dimensions=dimenR)
+
+        # Make the kernel image we are going to convolve.
+        scienceMI = self._makePsfMaskedImage(sciencePsfModel, center, dimensions=dimenR)
+
+        # The image to convolve is the science image, to the target Psf.
+        kc = diffimLib.makeKernelCandidate(center.getX(), center.getY(), scienceMI, targetMI, self.ps)
+        kernelCellSet.insertCandidate(kc)
+
+        return pipeBase.Struct(kernelCellSet=kernelCellSet,
+                               targetPsfModel=targetPsfModel,
+                               )
+
+    def _solve(self, kernelCellSet, basisList):
+        """Solve for the PSF matching kernel
+
+        Parameters
+        ----------
+        kernelCellSet : `~lsst.afw.math.SpatialCellSet`
+            A SpatialCellSet to use in determining the matching kernel
+            (typically as provided by _buildCellSet).
+        basisList : `list` [`~lsst.afw.math.kernel.FixedKernel`]
+            A sequence of Kernels to be used in the decomposition of the kernel
+            (typically as provided by makeKernelBasisList).
+
+        Returns
+        -------
+        spatialSolution : `~lsst.ip.diffim.KernelSolution`
+            Solution of convolution kernels.
+        psfMatchingKernel : `~lsst.afw.math.LinearCombinationKernel`
+            Spatially varying Psf-matching kernel.
+        backgroundModel : `~lsst.afw.math.Function2D`
+            Spatially varying background-matching function.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if unable to determine PSF matching kernel.
+        """
+        # Visitor for the single kernel fit.
+        if self.useRegularization:
+            singlekv = diffimLib.BuildSingleKernelVisitorF(basisList, self.ps, self.hMat)
+        else:
+            singlekv = diffimLib.BuildSingleKernelVisitorF(basisList, self.ps)
+
+        # Visitor for the kernel sum rejection.
+        ksv = diffimLib.KernelSumVisitorF(self.ps)
+
+        try:
+            # Make sure there are no uninitialized candidates as
+            # active occupants of Cell.
+            nRejectedSkf = -1
```

Comment:
Yes, we only have 1 kernel candidate. So it makes sense to remove the `while` loop. Would it make sense to remove the whole block as it is, since the `singlekv` kernel visitor's only purpose seems to retrieve `NRejected`?